ui/packages/catalog-ui-search/src/main/webapp/react-component/container/result-add/index.tsx

```diff
@@ -0,0 +1,131 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+import * as React from 'react'
+import { hot } from 'react-hot-loader'
+import withListenTo, { WithBackboneProps } from '../backbone-container'
+import * as _ from 'lodash'
+import * as ListModel from '../../../js/model/List'
+import * as store from '../../../js/store'
+
+import View from '../../presentation/result-add'
+
+const LIST_CQL_KEY = 'list.cql'
+const LIST_BOOKMARKS_KEY = 'list.bookmarks'
+const LIST_ICON_KEY = 'list.icon'
+
+type Props = {
+  model: [
+    {
+      matchesCql: (item: any) => any
+    }
+  ]
+} & WithBackboneProps
+
+type State = {
+  resultItems?: [any]
+}
+
+type List = {
+  toJSON: () => [any]
+}
+
+const formatListItems = (list: any, model: Props['model']) => {
+  const matchesFilter =
+    (list[LIST_CQL_KEY] !== '' &&
+      model.every(item => item.matchesCql(list[LIST_CQL_KEY]))) ||
+    true
+  const alreadyContains =
+    (_.intersection(
+      list[LIST_BOOKMARKS_KEY],
+      model.map((item: any) => item.get('metacard').id)
+    ).length === model.length &&
+      true) ||
+    false
+  const icon = ListModel.getIconMapping()[list[LIST_ICON_KEY]]
+
+  return {
+    ...list,
+    matchesFilter: matchesFilter,
+    alreadyContains: alreadyContains,
+    icon: icon,
+  }
+}
+
+const formatList = (model: Props['model']) => {
+  const json = getLists().toJSON()
+
+  return json.map((list: List) => formatListItems(list, model)) as [any]
+}
+
+const getLists = () => store.getCurrentWorkspace().get('lists')
+
+const addToList = (id: any, metacardIds: [any]) =>
+  getLists()
+    .get(id)
+    .addBookmarks(metacardIds)
+
+const removeFromList = (id: any, metacardIds: [any]) =>
+  getLists()
+    .get(id)
+    .removeBookmarks(metacardIds)
+
+export default hot(module)(
+  withListenTo(
+    class extends React.Component<Props, State> {
+      constructor(props: Props) {
+        super(props)
+
+        this.state = {
+          resultItems: formatList(this.props.model),
+        }
+      }
+
+      componentDidMount = () => {
```

Comment:
can you do this all in the constructor or is it necessary to do this in componentDidMount? I think this would be a good pattern to follow https://github.com/codice/ddf/pull/4458/files#diff-0598f5d2aa848d8fec1a22ef1e079296R69

Reply:
In my opinion a construct should create and return a component as soon as possible. Wiring up listeners and various asynchronous operations are only needed if it's mounted.